### PR TITLE
Replace rmmseg gem for cppjieba_rb since better dictionary

### PR DESCRIPTION
**Pull request by fantasticfears**

### CODE REVIEW

**Comment by fantasticfears** [comment]
New binding gem is here. @SamSaffron https://github.com/fantasticfears/cppjieba_rb

It's extracted from #4974.

File: app/services/search_indexer.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def self.update_index(table, id, raw_data)
     end
 
     # for user login and name use "simple" lowercase stemmer
-    stemmer = table == "user" ? "simple" : Search.long_locale
+    stemmer = table == "user" ? "simple" : Search.ts_config
```
Comment (SamSaffron) [inline app/services/search_indexer.rb]: hmm why was this change to use ts_config? I guess long_local may not even be available in pg?

Comment (fantasticfears) [inline app/services/search_indexer.rb]: PG took a text search configuration as the parameters for `TO_TSVECTOR` & `TO_TSQUERY`. It isn't related to locale so it's better for me to name with that intent.

From PG's doc:
```
CREATE TEXT SEARCH DICTIONARY english_ispell (
    TEMPLATE = ispell,
    DictFile = english,
    AffFile = english,
    StopWords = english
);
```

**Comment by SamSaffron** [comment]
OK, let's give this a shot! thanks!

**Pull request merged**